lib/archethic/p2p/node.ex

```diff
@@ -435,6 +302,13 @@ defmodule Archethic.P2P.Node do
     {origin_public_key, <<last_update_date::32, availability_update::32, rest::bitstring>>} =
       Utils.deserialize_public_key(rest)
 
+    {mining_public_key, rest} =
+      try do
+        Utils.deserialize_public_key(rest)
+      rescue
+        _ -> {nil, rest}
+      end
```

Comment:
Using a true/false byte avoid to use a try rescue
```suggestion
    {mining_public_key, rest} =
      case rest do
        <<1::8, rest::bitstring>> -> Utils.deserialize_public_key(rest)
        <<0::8, rest::bitstring>> -> {nil, rest}
      end
```